### DISCUSSION TITLE
LGPLv3 license

### DISCUSSION THREAD

**Discussion by jazzay** [nxrighthere/UnrealCLR#63] [Legacy]
This is a very interesting project for the UE4 development community. However I believe your selection of LGPL will prevent wider adoption of this awesome work, especially for commercial game development. 

Any chance you would consider changing the license to something friendlier?

**Comment by nxrighthere** [comment]
What problems do you have with LGPL from the perspective of a commercial project?

**Comment by jazzay** [comment]
From my experience no company legal team will allow (L)GPL licensed open source to be used within their commercial projects. I've seen some open source projects dual license, perhaps you want to go that route.

**Comment by schotime** [comment]
@jazzay  Do you intend to change the unrealclr code and not contribute the change back? 

Yes - that isn't very open source
No - you are free to use it.

**Comment by nxrighthere** [comment]
> From my experience no company legal team will allow (L)GPL licensed open source to be used within their commercial projects.

That makes sense when it comes to GPL license due to strong copyleft, but as for LGPL, I don't see any problems with it, especially with [permission](https://github.com/nxrighthere/UnrealCLR/blob/master/LICENSE#L169) to statically link compiled code with independent modules.

LGPL is about protecting the project from leeching. You have to contribute back to the open-source if you make any changes to code and you have to disclose your fork in such case. That's the main point.

**Comment by jazzay** [comment]
I did not notice this exception clause, because to be honest I did not read the license. To be clear I am no expert in the legalities of open source, all I know is from my experience in working in multiple public companies, they won't touch projects that even have a hint of (L)GPL. If you want to protect your work perhaps you should consider dual licensing, where companies can opt in to more compatible license terms.

**Comment by nxrighthere** [comment]
The plugin is now distributed under MIT license! Thanks Epic Games for [making it possible](https://twitter.com/nx_righthere/status/1356326412597293057)!

**Reply by jazzay** [reply]
This is great news. Congrats!
